spec/fixtures/rails_app/app/models/jwt_whitelist.rb

```diff
@@ -0,0 +1,6 @@
+class JwtWhitelist < ApplicationRecord
+  self.table_name = 'jwt_whitelist'
+
+  validate :jti, :aud, presence: true
+  validate :jti, uniqueness: true
```

Comment:
In fact there is no need to validate anything. `devise-jwt` should already be testing the behaviours, and the constraints (requirement and uniqueness) should be recommended to be added at the database level.

Reply:
@waiting-for-dev should the `aud` column be `null: false`?

Reply:
Yes, I would recommend to enforce that

Reply:
What about the case when the client doesn't provide the `JWT_AUD` header but this strategy is still used?

Reply:
Well, this strategy is meant to be used with client identifiers in place. For other cases it would be easy to implement a simple whitelist revocation strategy without client information.

By the way, it makes me thing that we should tell the strategy `ClientWhitelist` or something like that (I know I told about removing that from the nomenclature, but with your comments I see it is a better fit)

Reply:
However, notice that it is jut a recommendation for the user stated in the README, but she can actually do whatever she preffers.

Reply:
I like `ClientWhitelist` (or should it be `ClientsWhitelist` with plural `Clients`?) better.
Also, should i remove the `jwt_` prefix from the table?

Reply:
@pablocrivella , just now I was going to tell you that surely it is better not to worry about the renaming. I would let is as it is, `JwtWhitelist`, because it can be used in both ways: as a client whitelist using `aud` claim and as a general whitelist letting `aud` to nil.

Reply:
@waiting-for-dev Alright if i understand correctly i shoudl remove the `null: true` from the `aud` column right?

I'm a bit OCD with naming 😅  so i was already thinking it would be better to call the associated model `whitelisted_clients` or `whitelisted_tokens`
That way `has_many :whitelisted_xs` sounds more understandable than `has_many :jwt_whitelist`.

What do you think?

Reply:
> @waiting-for-dev Alright if i understand correctly i shoudl remove the null: true from the aud column right?

Well, it only affects to our tests, the end-user can create the migration as he wants. In our case, do as you preffer.

> I'm a bit OCD with naming :sweat_smile: so i was already thinking it would be better to call the associated model whitelisted_clients or whitelisted_tokens
That way has_many :whitelisted_(x)s sounds more understandable than has_many :jwt_whitelist.

Yes, `has_many :jwt_whitelists` sounds a bit awkward. To be more explicit, what about `:whitelisted_jwts`? We are already using something like `:jwt_blacklist`, so it makes sense.